mongodbatlas/fw_resource_mongodbatlas_alert_configuration.go

```diff
@@ -555,101 +555,100 @@ func (r *AlertConfigurationRS) ImportState(ctx context.Context, req resource.Imp
 
 	for i := range tfNotificationSlice {
-		value := tfNotificationSlice[i]
-
-		if value.IntervalMin.ValueInt64() > 0 {
-			typeName := value.TypeName.ValueString()
+		if !tfNotificationSlice[i].IntervalMin.IsNull() && tfNotificationSlice[i].IntervalMin.ValueInt64() > 0 {
+			typeName := tfNotificationSlice[i].TypeName.ValueString()
 			if strings.EqualFold(typeName, pagerDuty) || strings.EqualFold(typeName, opsGenie) || strings.EqualFold(typeName, victorOps) {
 				return nil, fmt.Errorf(`'interval_min' doesn't need to be set if type_name is 'PAGER_DUTY', 'OPS_GENIE' or 'VICTOR_OPS'`)
 			}
 		}
-
-		notifications[i] = matlas.Notification{
-			APIToken:                 value.APIToken.ValueString(),
-			ChannelName:              value.ChannelName.ValueString(),
-			DatadogAPIKey:            value.DatadogAPIKey.ValueString(),
-			DatadogRegion:            value.DatadogRegion.ValueString(),
-			DelayMin:                 pointy.Int(int(value.DelayMin.ValueInt64())),
-			EmailAddress:             value.EmailAddress.ValueString(),
-			EmailEnabled:             value.EmailEnabled.ValueBoolPointer(),
-			IntervalMin:              int(value.IntervalMin.ValueInt64()),
-			MobileNumber:             value.MobileNumber.ValueString(),
-			OpsGenieAPIKey:           value.OpsGenieAPIKey.ValueString(),
-			OpsGenieRegion:           value.OpsGenieRegion.ValueString(),
-			ServiceKey:               value.ServiceKey.ValueString(),
-			SMSEnabled:               value.SMSEnabled.ValueBoolPointer(),
-			TeamID:                   value.TeamID.ValueString(),
-			NotifierID:               value.NotifierID.ValueString(),
-			TypeName:                 value.TypeName.ValueString(),
-			Username:                 value.Username.ValueString(),
-			VictorOpsAPIKey:          value.VictorOpsAPIKey.ValueString(),
-			VictorOpsRoutingKey:      value.VictorOpsRoutingKey.ValueString(),
-			Roles:                    value.Roles,
-			MicrosoftTeamsWebhookURL: value.MicrosoftTeamsWebhookURL.ValueString(),
-			WebhookSecret:            value.WebhookSecret.ValueString(),
-			WebhookURL:               value.WebhookURL.ValueString(),
-		}
 	}
 
+	for i := range tfNotificationSlice {
+		notification := admin.AlertsNotificationRootForGroup{
+			ApiToken:                 tfNotificationSlice[i].APIToken.ValueStringPointer(),
+			ChannelName:              tfNotificationSlice[i].ChannelName.ValueStringPointer(),
+			DatadogApiKey:            tfNotificationSlice[i].DatadogAPIKey.ValueStringPointer(),
+			DatadogRegion:            tfNotificationSlice[i].DatadogRegion.ValueStringPointer(),
+			DelayMin:                 pointy.Int(int(tfNotificationSlice[i].DelayMin.ValueInt64())),
+			EmailAddress:             tfNotificationSlice[i].EmailAddress.ValueStringPointer(),
+			EmailEnabled:             tfNotificationSlice[i].EmailEnabled.ValueBoolPointer(),
+			IntervalMin:              pointy.Int(int(*tfNotificationSlice[i].IntervalMin.ValueInt64Pointer())),
```

Comment:
```suggestion
			IntervalMin:              util.Int64PtrToIntPtr(value.IntervalMin.ValueInt64Pointer()),
```
avoids risk of accessing the pointers value

Reply:
Changed it with your suggestion